File: mobile/src/application/usecases/getPendingFlows.ts
import { PendingFlow } from "@/domain"
import { PendingFlowPort } from "@/application/ports"
import { GetPendingFlows } from "@/domain/usecases"

export class GetPendingFlowsImpl implements GetPendingFlows {
  constructor(private pendingFlowPort: PendingFlowPort) {}

  async execute(): Promise<PendingFlow[]> {
    return this.pendingFlowPort.getAll()
  }
}

File: finanze/application/use_cases/get_pending_flows.py
from application.ports.pending_flow_port import PendingFlowPort
from domain.earnings_expenses import PendingFlow
from domain.use_cases.get_pending_flows import GetPendingFlows


class GetPendingFlowsImpl(GetPendingFlows):
    def __init__(self, pending_flow_port: PendingFlowPort):
        self._pending_flow_port = pending_flow_port

    async def execute(self) -> list[PendingFlow]:
        return await self._pending_flow_port.get_all()
